gspread/client.py

```diff
@@ -491,20 +491,26 @@ def insert_permission(
         """
 
         url = "{}/{}/permissions".format(DRIVE_FILES_API_V3_URL, file_id)
-
         payload = {
-            "emailAddress": value,
             "type": perm_type,
             "role": role,
             "withLink": with_link,
         }
-
         params = {
-            "sendNotificationEmail": notify,
-            "emailMessage": email_message,
             "supportsAllDrives": "true",
         }
 
+        if perm_type == "domain":
+            payload["domain"] = value
+        elif perm_type in {"user", "group"}:
+            payload["emailAddress"] = value
+            params["sendNotificationEmail"] = notify
+            params["emailMessage"] = email_message
```

Comment:
after reading the documentation of the drive API V3, the `emailMessage` is not restricted to only `user` or `groupe` type.
We can keep it in the general parameters. I don't know who will receive the email in case of `domain` :thinking: so keeping it like this make more sense to me.

Reply:
I had the same thought. If someone wants this, would be trivial to adjust later.